src/ast/passes/semantic_analyser.cpp

```diff
@@ -514,7 +519,48 @@ void SemanticAnalyser::visit(Call &call)
       auto &expr = (*call.vargs)[i];
       func_arg_idx_ = i;
 
-      expr->accept(*this);
+      // Handling offsetof() 2nd argument
+      // Handle the second parameter offsetof(record, field), this parameter
+      // is treated as an Identifier, of course, it may be call or builtin, we
+      // convert them all to Identifiers. For example:
+      //
+      //  struct foo {
+      //    int offsetof;   > offsetof(struct foo, offsetof)
+      //    int kstack;     > offsetof(struct foo, kstack)
+      //    int ctx;        > offsetof(struct foo, ctx)
+      //    int comm;       > offsetof(struct foo, comm)
+      //  }
+      if (func_ == "offsetof" && func_arg_idx_ == 1)
+      {
+        auto identifier = dynamic_cast<Identifier *>(call.vargs->at(1));
+        auto builtin = dynamic_cast<Builtin *>(call.vargs->at(1));
+        auto _call = dynamic_cast<Call *>(call.vargs->at(1));
+
+        if (identifier)
+        {
+          expr = new Identifier(identifier->ident, call.loc);
+          expr->accept(*this);
+        }
+        else if (builtin)
+        {
+          expr = new Identifier(builtin->ident, call.loc);
```

Comment:
Isn't this introducing a memleak? Who deletes the previous `Identifier`?

Reply:
I just `delete` the prev identifier, thanks for the review, please review again.